### ISSUE TITLE
[bitnami/rabbitmq] When using load-definition - username/password secret is not created

### ISSUE THREAD

**Issue by stephenpope** [bitnami/charts#7766]
**Which chart**:
`bitnami/rabbitmq` - Version: `8.22.4`

**Describe the bug**
When providing a load-definition (e.g to apply a default policy for all queues) the username and password secret is no longer created.

This makes it very hard to share credentials with other services that use RabbitMQ - I have to create my own username/password secret which is not a problem but seems odd that the functionality exists in the chart but is now disabled when load-definition is enabled.

I think the secret should still be created even if load-definition is enabled as it is a convenient way to share credentials with other applications.

**To Reproduce**

When `loadDefinition` is enabled then (for some reason) we have to provide a `user` section in order to make the solution work.

With a `values.yaml` configuration :

```yaml
rabbitmq:
  auth:
    username: "myuser"
    password: "yourStrong(!)Password"
  extraConfiguration: |-
    load_definitions = /app/load_definition.json
  extraSecrets:
    rabbitmq-load-definition:
      load_definition.json: |
        {
          "users": [
            {
              "name": "{{ .Values.auth.username }}",
              "password": "{{ .Values.auth.password }}",
              "tags": "administrator"
            }
          ],
          "permissions": [
            {
              "user": "{{ .Values.auth.username }}",
              "vhost": "/",
              "configure": ".*",
              "write": ".*",
              "read": ".*"
            }
          ],
          "vhosts": [
            {
              "name": "/"
            }
          ],
          "policies": [
            {
              "name": "lazy-queue",
              "vhost": "/",
              "pattern": ".*",
              "apply-to":"queues",
              "definition": {
                "queue-mode": "lazy"
              }
            }
          ]
        }
  loadDefinition:
    enabled: true
    existingSecret: rabbitmq-load-definition
```

The secret has explicit code so it is not created if `loadDefinition.enabled=true` ..

From: `rabbitmq/templates/secrets.yaml`

```yaml
#<snip>
data:
  {{- if and (not .Values.auth.existingPasswordSecret) (not .Values.loadDefinition.enabled) }}
  {{- if .Values.auth.password }}
  rabbitmq-password: {{ .Values.auth.password | b64enc | quote }}
  {{- else }}
  rabbitmq-password: {{ randAlphaNum 10 | b64enc | quote }}
  {{- end }}
  {{- end }}
  {{- if not 
#</snip>
``` 

If I want to share the password with a `3rd party application` in a secure manner I would reference it as such ..

```yaml
env:
  - name: RABBITMQ_PASSWORD
    valueFrom:
      secretKeyRef:
        name: release-rabbitmq
        key: rabbitmq-password~
```

The problem is that it is now **much harder** to do this as the only secret containing the username/password (in the example `rabbitmq-load-definition`) is contained within some `JSON`.

**Expected behavior**
If I have provided `auth.username` + `auth.password` I would expect the secret to still be created even if the `loadDefintion.eanbled: true`

TLDR: the `(not .Values.loadDefinition.enabled)` should be removed in `secrets.yaml`.

**Comment by amnk** [comment]
It is indeed not convenient (faced it myself), but since chart supports creating extra secrets, you can do something like this:
```
  extraSecrets:
    rabbitmq-auth-provider:
      rabbitmq-username: "{{ .Values.auth.username }}"
      rabbitmq-password: "{{ .Values.auth.password }}"
```

**Comment by martinounap** [comment]
And if I have created secred before what to use, then it is also not used.
